execution_engine/src/core/engine_state/step.rs

```diff
@@ -0,0 +1,97 @@
+use std::{collections::BTreeMap, fmt::Display, vec::Vec};
+
+use core::fmt;
+use uint::static_assertions::_core::fmt::Formatter;
+
+use casper_types::{bytesrepr, bytesrepr::ToBytes, Key, ProtocolVersion, PublicKey};
+
+use crate::shared::{newtypes::Blake2bHash, TypeMismatch};
+
+#[derive(Debug)]
+pub struct SlashItem {
+    pub validator_id: PublicKey,
+}
+
+impl SlashItem {
+    pub fn new(validator_id: PublicKey) -> Self {
+        Self { validator_id }
+    }
+}
+
+#[derive(Debug)]
+pub struct RewardItem {
+    pub validator_id: PublicKey,
+    pub value: u64,
+}
+
+impl RewardItem {
+    pub fn new(validator_id: PublicKey, value: u64) -> Self {
+        Self {
+            validator_id,
+            value,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct StepRequest {
+    pub parent_state_hash: Blake2bHash,
```

Comment:
This is the hash of the state on which the step will be executed, right? I'm asking because the name suggests it's the parent block's state hash, but if I understand correctly, that state will usually be modified by the deploys in the switch block before we get to executing this step, so the hash here will be different from the parent block's state hash.

Reply:
Changed the name.